nucliadb/src/nucliadb/train/generators/utils.py

```diff
@@ -29,29 +31,44 @@
 from nucliadb.train.types import T
 from nucliadb_utils.utilities import get_storage
 
-rcache: ContextVar[Optional[dict[str, ResourceORM]]] = ContextVar("rcache", default=None)
+
+class ResourceCache:
+    def __init__(self, size: int = 1024):
```

Comment:
I would set a lower size for the cache.

Bear in mind that in the resource object we're accumulating the extracted texts and field metadata behaviours, which can be 100s of MBs in some cases (large pdfs, for instance)

So I would rather get some cache misses than dying of OOM

Reply:
What do you think is a good value then? Right now we have an unbounded cache and 600Mi, so I don't think is fair to assume all resources are 100s of MBs